File: lab3/src/Matrix.java
import java.util.Arrays;
import java.util.Collections;

/**
 * Created by mamaxropela on 01.10.14.
 */
public class Matrix {
    private int[][] matrix;

    public Matrix(int size){
        matrix = new int[size][size];
    }

    public void Generate(int filledDigit){
        for (int i = 0; i < matrix.length; i++)
            for(int j = 0; j < matrix.length; j++)
                matrix[i][j] = filledDigit;
    }

    public static Matrix Multiply(Matrix m1, Matrix m2){
        Matrix rezult = new Matrix(m1.getSize());
        int tempValue = 0;
        for (int i = 0; i < rezult.getSize(); i++)
            for (int j = 0; j < rezult.getSize(); j++) {
                tempValue = 0;
                for (int k = 0; k < rezult.getSize(); k++)
                    tempValue += m1.getElement(j,k) * m2.getElement(k,j);
                rezult.setElement(i,j,tempValue);
            }
        return rezult;
    }

    public static Vector Multiply(Matrix matrix, Vector vector){
        Vector rezult = new Vector(vector.getSize());
        int tempValue = 0;
        for(int i = 0; i < rezult.getSize(); i++){
            tempValue = 0;
            for (int j = 0; j < rezult.getSize(); j++)
                tempValue += vector.getElement(i) * matrix.getElement(j,i);
            rezult.setElement(i,tempValue);
        }
        return rezult;
    }

    public static Matrix Plus(Matrix m1, Matrix m2){
        Matrix rezult = new Matrix(m1.getSize());
        for (int i = 0; i < m1.getSize(); i++)
            for (int j = 0; j < m1.getSize(); j++)
                rezult.setElement(i,j,m1.getElement(i,j) + m2.getElement(i,j));
        return rezult;
    }

    public Matrix Sort(){
        Integer[] tempArr = new Integer[matrix.length];
        for (int i = 0; i < matrix.length; i++){
            for (int j = 0; j < matrix.length; j++)
                tempArr[j] = matrix[i][j];
            Arrays.sort(tempArr,Collections.reverseOrder());
            for(int j = 0; j < matrix.length; j++)
                matrix[i][j] = tempArr[j];
        }
        return this;
    }

    public Matrix Transpose(){
        Matrix rezult = new Matrix(matrix.length);
        for (int i = 0; i < matrix.length; i++)
            for(int j = 0; j < matrix.length; j++)
                rezult.setElement(i,j,matrix[j][i]);
        return rezult;
    }

    public int getSize(){
        return matrix.length;
    }

    public int getElement(int line, int column){
        return matrix[line][column];
    }
    public void setElement(int line, int column, int value){
        matrix[line][column] = value;
    }

}

File: 2 half/lab8/lab8/Matrix.cpp
#include "Data.h"
#include <algorithm>
#include <iostream>

Matrix::Matrix(int sizes){
	size = sizes;
	matrix = new int*[size];
	int *alloc = new int[sizes*sizes];
	for (int i = 0; i < sizes; i++)
		matrix[i] = &(alloc[sizes*i]);
}

void Matrix::generate(int filledDigit){
	for (int i = 0; i < size; i++)
	for (int j = 0; j < size; j++)
		matrix[i][j] = filledDigit;
}

Matrix* Matrix::operator*(Matrix &matr){
	Matrix *rez = new Matrix(size);
	int tempValue = 0;
	for (int i = 0; i < size; i++)
	for (int j = 0; j < size; j++) {
		tempValue = 0;
		for (int k = 0; k < size; k++)
			tempValue += matrix[j][k] * matr.getElement(k, j);
		rez->setElement(i, j, tempValue);
	}
	return rez;
}

void Matrix::print(){
	for (int i = 0; i < size; i++){
		for (int j = 0; j < size; j++)
			std::printf("%6d  ", matrix[i][j]);
		std::printf("\n");
	}
}

Matrix* Matrix::operator+(Matrix &matr){
	Matrix *rezult = new Matrix(size);
	for (int i = 0; i < size; i++)
	for (int j = 0; j < size; j++)
		rezult->setElement(i, j, matrix[i][j] + matr.getElement(i, j));
	return rezult;
}

int compare(const void *a, const void *b)
{
	return *((int*)(b)) - *((int*)a);
}

Matrix* Matrix::sort(){
	Matrix *rezult = new Matrix(size);
	std::copy(matrix, static_cast<int**>(matrix) + size, rezult->matrix);
	for (int i = 0; i < size; i++)
		qsort(rezult->matrix[i], size, sizeof(int), &compare);
	return rezult;
}

Matrix* Matrix::transpose(){
	Matrix *rezult = new Matrix(size);
	for (int i = 0; i < size; i++)
	for (int j = 0; j < size; j++)
		rezult->setElement(i, j, matrix[j][i]);
	return rezult;
}

int Matrix::getSize(){
	return size;
}

int Matrix::getElement(int line, int column){
	return matrix[line][column];
}
void Matrix::setElement(int line, int column, int value){
	matrix[line][column] = value;
}

Matrix::~Matrix()
{
	delete[] matrix[0];
	delete[] matrix;
}

Vector* operator*(Matrix &matrix, Vector &vector){
	Vector *rez = new Vector(vector.getSize());
	int tempValue = 0;
	for (int i = 0; i < vector.getSize(); i++){
		tempValue = 0;
		for (int j = 0; j < vector.getSize(); j++)
			tempValue += vector.getElement(i) * matrix.getElement(j, i);
		rez->setElement(i, tempValue);
	}
	return rez;
}

Vector* operator*(Vector &vector, Matrix &matrix){
	Vector *rez = new Vector(vector.getSize());
	int tempValue = 0;
	for (int i = 0; i < vector.getSize(); i++){
		tempValue = 0;
		for (int j = 0; j < vector.getSize(); j++)
			tempValue += vector.getElement(i) * matrix.getElement(j, i);
		rez->setElement(i, tempValue);
	}
	return rez;
}